OptimoveCore/Sources/Classes/NetworkClient/NetworkClient.swift

```diff
@@ -4,6 +4,23 @@ import Foundation
 
 public typealias NetworkServiceCompletion = (Result<NetworkResponse<Data?>, NetworkError>) -> Void
 
+extension NetworkClient {
```

Comment:
there are major discrepancies VS android:

1. tenantId, region and brandGroupId are taken from base64 encoded string, not from storage set by other features which may not exist
2. this base64 encoded string is set onto config during initialisation, not passed into API calls
3. calling enablePreferenceCenter on config enables the PC feature
4. here customer Id is not checked for being null or being a visitor id
5. in android set/get preferences gives you back a result enum
6. API sits not in Optimove, but in OptimovePreferenceCenter, similar to OptimoveInApp
7. here API calls can throw / break execution and doesnt log errors VS in android we only log errors and give you respective Result into your callback
8. get/set API receives a callback as an argument VS async/await. Async/await are probably preferable in swift, but i think consistency wins. 
9. in android fetching happens on a bg thread, callbacks run on main thread. Here fetching happens on a random thread (one that is available?) it continues on the thread from which it was called

Some stuff is very debatable here, but I think we should update most (all?) for consistency

Reply:
comment on consistency. on slack i was posting

```
 // 0. initialisation
    //    Optimove.initialize(this, builder)
    //            ...
    //            .enablePreferenceCenter(brandGroupId)
    //            .build();
    // 1. API
     OptimovePreferenceCenter.getInstance().getPreferencesAsync(PreferencesHandler);
     OptimovePreferenceCenter.getInstance().setCustomerPreferencesAsync(PreferencesHandler, List<PreferenceUpdate> updates);
    //    public interface PreferencesGetHandler {
    //     void run(@Nullable Preferences preferences);
    //    }
    //    public interface PreferencesSetHandler {
    //      void run(Boolean result);
    //    }
    // 3. models matching interfaces in web sdk
    // 4. what happens if no customer id? getPreferencesAsync, setCustomerPreferencesAsync immediately return, log warning, dont crash
```

which has for example
`OptimovePreferenceCenter.getInstance().setCustomerPreferencesAsync(PreferencesHandler, List<PreferenceUpdate> updates);`

vs

here we have 
`func setPreferences(for customerId: String, brandGroupId: String, _ preferenceUpdates: [PreferenceUpdate]) async throws -> Preferences`

the differences being:
1. it shouldnt take customer id
2. it shouldnt take brand group it
3. it should take callback
4. it should be called on OptimovePreferenceCenter
5. it's called differently

The point i'm trying to make is not that the interface i was proposing makes most sense (it may be total garbage), but that we should sync on the desired stuff, argue if it doesnt make --> produce smth more consistent from start --> win 🎱